packages/entities/entities-plugins/src/components/PluginEntityForm.vue

```diff
@@ -0,0 +1,717 @@
+
+      let fieldObject: Record<string, any> = {}
+
+      // Rather then over writting the key in output model we, use the existing object if there is one
+      // this allows us to do things like support multiple developer meta fields
+      if (outputModel[fieldNameDotNotation]) {
+        fieldObject = outputModel[fieldNameDotNotation]
+      }
+
+      fieldObject[key] = fieldValue
+      fieldValue = fieldObject
+    }
+
+    // Empty fields are set to null to tell Kong to remove the value
+    // or reset this value to it's default.
+
+    // Include if field is empty & field has been modified from orginal value
+    if (originalValue !== undefined && fieldValue === '' && fieldValue !== originalValue) {
+      // We need to determine what type of 'empty' value to set for modified nested inputs,
+      // if we do not the model will not preserve its structure
+      // (this change may be able to be removed when core updates its reset default handling)
+      outputModel[fieldNameDotNotation] = fieldSchemaValueType === 'object' ? {} : null
+
+      // If empty & field is a checklist array, set as [] to prevent all options being selected
+    } else if (fieldSchema.type === 'checklist' && fieldValue === '') {
+      outputModel[fieldNameDotNotation] = []
+
+      // Include input value if field is not empty
+    } else if (fieldValue !== '') {
+      outputModel[fieldNameDotNotation] = fieldValue
+    }
+
+    // Do an optional final transform, as defined in transform() function in schema field
+    outputModel[fieldNameDotNotation] = transformer
+      ? transformer(fieldValue)
+      : outputModel[fieldNameDotNotation]
+
+    // If dot notated key (service.id) is null, set the whole object to null (service = null)
+    unsetNullForeignKey(fieldNameDotNotation, outputModel)
+  })
+
+  return unFlattenObject(outputModel)
+}
+
+const loading = ref(true)
+const initFormModel = (): void => {
+  if (props.record && props.schema) {
+    // top level fields
+    updateModel({
+      enabled: props.record.enabled ?? true,
+      ...(props.record.instance_name && { instance_name: props.record.instance_name || '' }),
+      ...(props.record.protocols && { protocols: props.record.protocols }),
+      ...(props.record.tags && { tags: props.record.tags }),
+    })
+
+    if (props.record.data) {
+      const newModel = props.record.data
+      newModel.client_certificate = newModel.client_certificate?.id
+      updateModel(props.record.data)
+    } else if (props.record.config) {
+      if (props.record.consumer_id || props.record.service_id || props.record.route_id || props.record.consumer_group_id) {
+        updateModel({
+          service_id: props.record.service_id,
+          route_id: props.record.route_id,
+          consumer_id: props.record.consumer_id,
+          consumer_group_id: props.record.consumer_group_id,
+          enabled: props.record.enabled,
+        })
+      }
+
+      updateModel(props.record.config, 'config')
+    }
```

Comment:
question: what is this block doing?

Reply:
I checked the file history and these lines can be removed:
```diff
    if (props.record.data) {
-      const newModel = props.record.data
-      newModel.client_certificate = newModel.client_certificate?.id
      updateModel(props.record.data)
    } else if (props.record.config) {
      if (props.record.consumer_id || props.record.service_id || props.record.route_id || props.record.consumer_group_id) {
        updateModel({
          service_id: props.record.service_id,
          route_id: props.record.route_id,
          consumer_id: props.record.consumer_id,
          consumer_group_id: props.record.consumer_group_id,
          enabled: props.record.enabled,
        })
      }

      updateModel(props.record.config, 'config')
    }
```

They are from [this PR](https://github.com/Kong/kong-admin/pull/2031/files#diff-67c30c7f791df36e84579a8d81db0c0bd180c11ade995771b87c3656a3528d7bR447) and have nothing to do with plugins.